### PR TITLE
Remove par type

### PR DESCRIPTION
**Pull request by thisac**
Removes the par-type from Blackbird.

### CODE REVIEW

File: blackbird_python/blackbird/listener.py [hunk 1]
```diff
@@ -278,7 +276,7 @@ def exitExpressionvar(self, ctx: blackbirdParser.ExpressionvarContext):
         try:
             # assume all variables are scalar
             final_value = PYTHON_TYPES[vartype](value)
-        except:
+        except TypeError:
```
Comment (josh146) [inline blackbird_python/blackbird/listener.py:279]: good catch!

Comment (josh146) [reply blackbird_python/blackbird/listener.py:279]: I wonder why I originally had this as a general except... perhaps there are some edge cases where something other than a `TypeError` can be raised?

File: blackbird_python/blackbird/tests/test_listener.py [hunk 2]
```diff
@@ -336,16 +336,6 @@ def test_operation_mode_expressions(self, parse_input_mocked_metadata):
         )
         assert bb.operations == [{'op': 'MZgate', 'args': [0, 1], 'kwargs': {}, 'modes': [8, 3]}]
 
-    @pytest.mark.parametrize("arr", ["\t{phase_0}\n\t{phase_1}\n", "\t{phase_0}, {phase_1}\n"])
-    def test_parameter_idx(self, arr, parse_input_mocked_metadata):
-        """Test that parameter array indexing works inside arguments"""
```
Comment (josh146) [inline blackbird_python/blackbird/tests/test_listener.py:341]: How come this test was deleted?

**Review by josh146 — APPROVED** [review]
Thanks @thisac!

File: blackbird_python/blackbird/listener.py (hunk 1 above)
Comment (thisac) [reply blackbird_python/blackbird/listener.py:279]: I was thinking about this as well, but couldn't figure out any other cases. Either way, I think it's better to add these cases separately if they show up, instead of having a catch-all except statement.

Comment (josh146) [reply blackbird_python/blackbird/listener.py:279]: agreed

File: blackbird_python/blackbird/tests/test_listener.py (hunk 2 above)
Comment (thisac) [reply blackbird_python/blackbird/tests/test_listener.py:341]: Correct me if I'm wrong, but the way I understand it a BB script with free parameters (i.e. a template script) should never be run directly, and thus it's not necessarily possible to access free parameters before specific parameter values have been assigned, as is done in this test. 

Instead I guess we could add further tests that fill in the template and _then_ test whether things works the way they should.

Comment (josh146) [reply blackbird_python/blackbird/tests/test_listener.py:341]: Got it! I didn't spot that the data in the parametrize decorator where templated parameters 🙂
